documentation/project-docs/breaking-change-guidelines.md

```diff
@@ -0,0 +1,81 @@
+# SDK Breaking Change and Diagnostic Guidelines
+
+This document provides guidelines for introducing new diagnostics or breaking changes to the .NET SDK, which configuration knobs are available, and what criteria should guide the decision around severity, timeline for introduction, and deprecation of diagnostics.
+
+## Overall procedure
+
+In general, we want to make updating the .NET SDK as smooth as possible for developers. This means:
+
+* introducing new changes in a staged/gradual way
+* trying to tie opinionated analyzers/diagnostics to a mechanism that requires explicit user opt-in
+* providing a way to opt out of a change entirely
+
```

Comment:
nit: The other bulleted lists begin with a capital letter. I prefer capital for lists and would suggest doing that here.

Reply:
Changed.